test/test_jit_autocast.py

```diff
@@ -819,6 +819,32 @@ def test_nhwc_autocast_jit_trace_model(model, x):
                 continue
             test_nhwc_autocast_jit_trace_model(self.models[i], self.inputs[i])
 
+    def test_cat_promote(self):
+        class TestModel(torch.nn.Module):
+            def __init__(self):
+                super(TestModel, self).__init__()
+
+            def forward(self, a, b):
+                return torch.cat([a, b], 0)
+        texpr_fuser_state = torch._C._jit_texpr_fuser_enabled()
+        torch._C._jit_set_texpr_fuser_enabled(False)
```

Comment:
Do you want to put the save and restore operation into a context manager to guarantee the restore of texpr_fuser flag always happens? Also, please add a comment on why you want to disable NNC here.

Reply:
Sure, use the context API and add comment for why disable NNC here.